### PR TITLE
Dev jokes js to ts

### LINKED ISSUE #563: ✨ [Feature request]:  Dev jokes javascript to typescript
**Is your feature request related to a problem? Please describe.**

This feature is not related to any problem it's modification of Dev jokes from javascript to typescript 

**Describe the solution you'd like**

Converting jsx to tsx 🛠️

### PR DESCRIPTION
**Pull request by yung-coder**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Here i have tried to change dev jokes from JavaScript to typescript .

Fixes #563 

## Type of change

- [ ✅] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
 It has been tested by me in my local system for all edge cases.

# Checklist:

- [ ✅] I have performed a self-review of my own code
- [✅ ] I have commented my code, particularly in hard-to-understand areas
- [✅ ] I have made corresponding changes to the documentation
- [✅ ] My changes generate no new warnings
- [ ✅] I have added tests that prove my fix is effective or that my feature works
- [✅ ] New and existing unit tests pass locally with my changes
- [ ✅] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by yung-coder** [comment]
hey @koustov  please give it a look ...

**Comment by Sachin-chaurasiya** [comment]
@yung-coder , Can you add the meaningful title for the PR?

File: src/plays/dev-jokes/DevJokes.tsx [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
+} from "react";
+import React from "react";
 import Spinner from "./Spinner";
-function DevJokes(props) {
-  const [joke, setjoke] = useState([]);
-  const [spinner, setspinner] = useState(false);
+
+
+function DevJokes(props: any) {
+  const [joke, setjoke] = useState<any[]>([]);
```
Comment (Sachin-chaurasiya) [inline src/plays/dev-jokes/DevJokes.tsx:13]: Can we use proper type for joke?
We should avoid using the `any`.

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]

File: src/plays/dev-jokes/DevJokes.tsx (hunk 1 above)
Comment (yung-coder) [reply src/plays/dev-jokes/DevJokes.tsx:13]: yeah i have tried to do same  i  have created a interface 
code:-
interface jokes{
  punchline: string
  question: string
}

but the problem is the key it throws err
pr
or

Comment (Sachin-chaurasiya) [reply src/plays/dev-jokes/DevJokes.tsx:13]: Value is object right and key should be number or string.
Can you try to use string or number for key?

Comment (yung-coder) [reply src/plays/dev-jokes/DevJokes.tsx:13]: I have tried it before it was not working...

Comment (Sachin-chaurasiya) [reply src/plays/dev-jokes/DevJokes.tsx:13]: What code you have tried can you share it?

Comment (yung-coder) [reply src/plays/dev-jokes/DevJokes.tsx:13]: This 

pr-modufy

Comment (Sachin-chaurasiya) [reply src/plays/dev-jokes/DevJokes.tsx:13]: @yung-coder, did you check what is the structure of API data? does it return the key attribute?

Comment (Sachin-chaurasiya) [reply src/plays/dev-jokes/DevJokes.tsx:13]: It only returns two attributes, `question` and `punchline`.
image

Comment (Sachin-chaurasiya) [reply src/plays/dev-jokes/DevJokes.tsx:13]: Your interface should only contain these attributes.

Comment (Sachin-chaurasiya) [reply src/plays/dev-jokes/DevJokes.tsx:13]: For the key, you can access the index value from the map callback function like this.

```ts
{joke.map((value,index) => {
                  return (
                    <div key={index}>
                      <h1>{value.question}</h1>
                      <p>{value.punchline}</p>
                    </div>
                  );
                })}
```

Comment (yung-coder) [reply src/plays/dev-jokes/DevJokes.tsx:13]: Thanks @Sachin-chaurasiya got it

**Comment by yung-coder** [comment]
hey @Sachin-chaurasiya  updated give it a look

File: src/plays/dev-jokes/DevJokes.tsx [hunk 2]
```diff
@@ -8,15 +8,21 @@ import {
...
 function DevJokes(props: any) {
-  const [joke, setjoke] = useState<any[]>([]);
+  const [joke, setjoke] = useState<jokes[]>([]);
   const [spinner, setspinner] = useState<boolean>(false);
   const fetch = () => {
     setspinner(true);
     axios
       .get("https://backend-omega-seven.vercel.app/api/getjoke")
       .then((response) => {
+        console.log(response.data)
```
Comment (Sachin-chaurasiya) [inline src/plays/dev-jokes/DevJokes.tsx:25]: Can we remove the console log?

Comment (yung-coder) [reply src/plays/dev-jokes/DevJokes.tsx:25]: Ohh sorry i Just forgot wait a sec

**Comment by yung-coder** [comment]
Done

File: src/plays/dev-jokes/DevJokes.tsx [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
+  useEffect,
+  useState,
+} from "react";
+import React from "react";
 import Spinner from "./Spinner";
-function DevJokes(props) {
-  const [joke, setjoke] = useState([]);
-  const [spinner, setspinner] = useState(false);
+
+interface jokes{
```
Comment (Sachin-chaurasiya) [inline src/plays/dev-jokes/DevJokes.tsx:11]:
```suggestion
interface Jokes{
```
@yung-coder , Interface name should start with a capital.

File: src/plays/dev-jokes/DevJokes.tsx [hunk 4]
```diff
@@ -1,11 +1,22 @@
...
+  punchline: string
+  question: string
+}
+
+
+function DevJokes(props: any) {
+  const [joke, setjoke] = useState<jokes[]>([]);
```
Comment (Sachin-chaurasiya) [inline src/plays/dev-jokes/DevJokes.tsx:18]:
```suggestion
  const [joke, setjoke] = useState<Jokes[]>([]);
```

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Requested minor change

**Comment by yung-coder** [comment]
Should I change the name of interface @Sachin-chaurasiya ..?

**Comment by Sachin-chaurasiya** [comment]
> Should I change the name of interface @Sachin-chaurasiya ..?

@yung-coder , Yes, please see the suggested changes.

**Comment by yung-coder** [comment]
done @Sachin-chaurasiya

File: src/plays/dev-jokes/DevJokes.tsx [hunk 5]
```diff
@@ -1,11 +1,22 @@
 import PlayHeader from "common/playlists/PlayHeader";
 import "./devJokes.css";
 import axios from "axios";
-import { useEffect, useState } from "react";
+import {
+  useEffect,
+  useState,
+} from "react";
+import React from "react";
```
Comment (Sachin-chaurasiya) [inline src/plays/dev-jokes/DevJokes.tsx:8]:
```suggestion
import React, {
  useEffect,
  useState,
} from "react";
```

File: src/plays/dev-jokes/DevJokes.tsx [hunk 6]
```diff
@@ -1,11 +1,22 @@
...
-  const [spinner, setspinner] = useState(false);
+
+interface Jokes{
+  punchline: string
+  question: string
+}
+
+
+function DevJokes(props: any) {
+  const [joke, setjoke] = useState<Jokes[]>([]);
```
Comment (Sachin-chaurasiya) [inline src/plays/dev-jokes/DevJokes.tsx:18]:
```suggestion
  const [jokes, setjokes] = useState<Jokes[]>([]);
```

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Few more comments.

**Comment by yung-coder** [comment]
done @Sachin-chaurasiya

File: src/plays/dev-jokes/DevJokes.tsx [hunk 7]
```diff
@@ -1,17 +1,23 @@
...
-  const [joke, setjoke] = useState([]);
-  const [spinner, setspinner] = useState(false);
+
+interface Jokes {
+  punchline: string;
+  question: string;
+}
+
+function DevJokes(props: any) {
+  const [joke, setJokes] = useState<Jokes[]>([]);
```
Comment (Sachin-chaurasiya) [inline src/plays/dev-jokes/DevJokes.tsx:13]: @yung-coder , `joke` should be `jokes` as it is an list.

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]

**Comment by yung-coder** [comment]
done @Sachin-chaurasiya  i think its all good now

**Review by Sachin-chaurasiya — APPROVED** [review]

**Comment by yung-coder** [comment]
hey @atapas  can you review this ..?

**Pull request merged**